File: backend/src/main/java/edu/cit/custodio/mdqueue/feature/auth/dto/AuthResponse.java
package edu.cit.custodio.mdqueue.feature.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Authentication response DTO.
 * <p>
 * Uses the <b>Builder Design Pattern</b> (via Lombok {@code @Builder}) to provide
 * a fluent, readable API for constructing response objects with multiple optional fields.
 * This avoids telescoping constructors and ensures immutability after construction.
 * </p>
 *
 * @see edu.cit.custodio.mdqueue.feature.auth.AuthService#buildAuthResponse
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {

    private String token;
    private String type;
    private Long userId;
    private String email;
    private String fullName;
    private String role;
    private String message;
}

File: mobile/app/src/main/java/edu/cit/custodio/mdqueue/api/models/AuthResponse.kt
package edu.cit.custodio.mdqueue.api.models

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    @SerializedName("token")
    val token: String?,

    @SerializedName("type")
    val type: String?,

    @SerializedName("userId")
    val userId: Long?,

    @SerializedName("email")
    val email: String?,

    @SerializedName("fullName")
    val fullName: String?,

    @SerializedName("message")
    val message: String?
)
